lib/helpers/get-repo-info.js

```diff
@@ -21,62 +21,61 @@ var getConfig = require('./get-config');
 // git@try.gogs.io:<org>/<project>.git
 // https://try.gogs.io/<org>/<project>.git
 
+module.exports = function getRepoInfo(project) {
+  return getRepoURL(project).then(function(link) {
+    var parts = link.split('/');
+    var projectName = parts.pop();
+    var org = parts.pop();
+    var service = getService(link);
+    if (projectName.indexOf('.git') !== -1) {
+      projectName = projectName.substr(0, projectName.indexOf('.git'));
+    }
+
+    return {
+      path: link,
+      organization: org,
+      name: projectName,
+      project: projectName,
+      service: service,
+      projectUrl: getUrlBase(link, service, org, projectName),
+      repositoryLink: link
+    };
+  });
+};
 
-module.exports = function() {
-  var pkg = readPackage.bind(this)();
-  var firstStep;
+function getRepoURL(project) {
+  var pkg = readPackage(project.root);
 
   if (!pkg.repository || (typeof pkg.repository !== 'string' && !pkg.repository.url)) {
-    firstStep = promptForRepoName.call(this)
+    return promptForRepoName(project)
       .then(function(repositoryLink) {
-        updatePackage.call(this, {
+        updatePackage(project, {
           repository: repositoryLink
         });
         return repositoryLink;
-      }.bind(this));
+      });
   } else {
-    firstStep = Promise.resolve(pkg.repository.url || pkg.repository);
+    return Promise.resolve(pkg.repository.url || pkg.repository);
   }
+}
 
-  return firstStep
-    .then(function(link) {
-      var parts = link.split('/');
-      var project = parts.pop();
-      var org = parts.pop();
-      var service = getService(link);
-      if (project.indexOf('.git') !== -1) {
-        project = project.substr(0, project.indexOf('.git'));
-      }
-      var p = {
-        root: this.project.root,
-        path: link,
-        organization: org,
-        name: project,
-        project: this.project,
-        version: pkg.version,
-        service: service,
-        projectUrl: getUrlBase(link, service, org, project),
-        repositoryLink: link
-      };
-      var config = getConfig.call(this, p);
-
-      config.hostIsGithub = p.service === 'github';
-      config.hostIsBitbucket = p.service === 'bitbucket';
-      config.hostIsCustom = p.service === 'custom';
-
-      config.project = p;
-      return config;
-    }.bind(this));
-};
```

Comment:
It looks as though you've removed the "service" detection without replacing it elsewhere.